src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -558,6 +558,16 @@ private void finishModCreation(boolean closeTab) {
 	protected abstract AggregatedValidationResult validatePage(int page);
 
 	protected void beforeGeneratableElementGenerated() {
+		// we delete mod element list templates (if any) to prevent outdated mod files from being present
+		// in case size of a list have changed
+		if (editingMode) {
```

Comment:
Handling of outdated files should be handled in the backend, not in the UI

Reply:
Should I move this to `ModElementManager#storeModElement()` that replaces the previous mod element definition with a new one?

Reply:
The best would be where existing deletions of eg. condition "deleteWhenConditionFalse" is done, so all deletions with this intent are done there

Reply:
List templates need the mod element to detokenize paths of output files, so if this is done after running the current code of `ModElementManager.storeModElement()`, the outdated files can no longer be accessed.

Reply:
The thing is, system needs to be designed in a way generation and deletion of the files is handled by the Generator system. ModElementManager takes care of storing mod element definition, not handling generator files

Reply:
I just thought to something. What if instead of this piece of code we call
```java
		mcreator.getGenerator().removeElementFilesAndLangKeys(element.getModElement());
```
before storing the new variant of the mod element being saved? This would also make `deleteWhenConditionFalse` parameter obsolete because all the files are anyway to be recreated in this case.

Reply:
> saving the mod element being edited

This still delegates generator tasks to mod element system, which is not ok. Generator should be as self contained as possible

Reply:
I didn't mean involving ModElementManager. I pushed what I meant right now.

Reply:
> This would also make deleteWhenConditionFalse parameter obsolete because all the files are anyway to be recreated each time in this case.

This is not a bad idea actually, just the location where it should be done needs to be right. But this is for another PR if you are interested in making this system more clean

Reply:
>But this is for another PR

The whole mod element JSON definition cache thing? If so, I'm delaying this PR once again because that would be needed here.

Reply:
> This would also make deleteWhenConditionFalse parameter obsolete because all the files are anyway to be recreated each time in this case.

This